citro3d/src/lib.rs

```diff
@@ -161,6 +162,44 @@ impl Instance {
             );
         }
     }
+    /// Indexed drawing
+    ///
+    /// Draws the vertices in `buf` indexed by `indices`. `indices` must be linearly allocated
+    ///
+    /// # Safety
+    /// If `indices` goes out of scope before the current frame ends it will cause a use-after-free (possibly by the GPU)
+    /// If `buf` does not contain all the vertices references by `indices` it will cause an invalid access by the GPU (this crashes citra)
```

Comment:
I think I'm fine with having a new `unsafe` API for now, but definitely would like to see if we can solve this with the type system somehow. I mentioned a possible option for the second issue in my other comment and I think the first issue could be solved with other stuff for #41 so I think we at least have a path forward.